client/wc-api/categories/selectors.js

```diff
@@ -0,0 +1,59 @@
+/** @format */
+
+/**
+ * External dependencies
+ */
+import { isNil } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import { getResourceName } from '../utils';
+import { DEFAULT_REQUIREMENT } from '../constants';
+
+const getCategories = ( getResource, requireResource ) => (
+	query = {},
+	requirement = DEFAULT_REQUIREMENT
+) => {
+	const resourceName = getResourceName( 'category-query', query );
+	const ids = requireResource( requirement, resourceName ).data || [];
+	const categories = ids.reduce(
+		( acc, id ) => ( {
+			...acc,
+			[ id ]: getResource( getResourceName( 'category', id ) ).data || {},
+		} ),
+		{}
+	);
+	return categories;
+};
+
+const getCategoriesTotalCount = ( getResource, requireResource ) => (
+	query = {},
+	requirement = DEFAULT_REQUIREMENT
+) => {
+	const resourceName = getResourceName( 'category-query', query );
+	return requireResource( requirement, resourceName ).totalCount || 0;
```

Comment:
Just to check, this will trigger a fetch if it's not available. Is that what you want here? If not, maybe switch this to `getResource` instead.

Reply:
Gotcha- thanks!  Updated to use `getResource`.